commit e61f27f56bef0f229f0bf02911f76cfcfa12f083
Author: Peter Volkov <pva@gentoo.org>
Date:   Wed Jul 8 16:06:18 2009 +0400

    Add --without-libnl configure switch
    
    Allow build libpcap with libnl disabled even in case libnl is installed
    at system.

diff --git a/configure.in b/configure.in
index 88a71ef..397a9c7 100644
--- a/configure.in
+++ b/configure.in
@@ -437,9 +437,19 @@ linux)
 	#
 	# Do we have libnl?
 	#
-	AC_CHECK_LIB(nl, nl_handle_alloc,
-		LIBS="-lnl $LIBS"
-		AC_DEFINE(HAVE_LIBNL,1,[if libnl exists]),)
+	AC_ARG_WITH(libnl,
+	AC_HELP_STRING([--without-libnl],[disable libnl support @<:@default=yes, on Linux, if present@:>@]),
+		with_libnl=$withval,,)
+
+	if test x$with_libnl != xno ; then
+		AC_CHECK_LIB(nl, nl_handle_alloc,
+			LIBS="-lnl $LIBS"
+			AC_DEFINE(HAVE_LIBNL,1,[if libnl exists]),
+			if test x$with_libnl = xyes ; then
+				AC_MSG_ERROR([libnl support requested but libnl not found])
+			fi
+		)
+	fi
 
 	AC_LBL_TPACKET_STATS
 	AC_LBL_LINUX_TPACKET_AUXDATA_TP_VLAN_TCI
